rclcpp/include/rclcpp/node.hpp

```diff
@@ -139,7 +139,7 @@ class Node : public std::enable_shared_from_this<Node>
   const std::vector<rclcpp::callback_group::CallbackGroup::WeakPtr> &
   get_callback_groups() const;
 
-  /// Create and return a Publisher.
+  /// Create and return a Publisher. Note: This constructor is deprecated
```

Comment:
Does this cause a warning from the compiler if the deprecated function is used? If so it would cause the builds to be unstable until we got all the usages switched over.

Reply:
That is my primary concern with that as well

Reply:
That's a good point.

Wondering if something like that could be a solution:

```c++
// TODO(ros2/rmw#1234): replace the following line by
// # define RMW_DEPRECATED(x) [[deprecated(x)]]
// on 2019-XX-YY.
# define RMW_NODE_CTOR_DEPRECATED(x)
```

and switch later to:

```c++
// TODO(ros2/rmw#56768): remove all code annotated with this on 20XX-YY-ZZ.
# define RMW_NODE_CTOR_DEPRECATED(x) [[deprecated(x)]]
```